backend/src/v5/models/revisions.js

```diff
@@ -45,4 +67,25 @@ Revisions.getRevisionCount = (teamspace, model) => {
 	return db.count(teamspace, collectionName(model), query);
 };
 
+Revisions.getRevisions = (teamspace, model, showVoid, projection = {}) => {
+	const query = { ...excludeIncomplete };
+
+	if (!showVoid) {
+		query.void = { $ne: true };
+	}
+
+	return findRevisionsByModel(teamspace, model, query, projection);
+};
+
+Revisions.updateRevisionStatus = async (teamspace, model, revision, status) => {
+	const rev = await findOneRevisionByID(teamspace, model, revision, { _id: 1, void: 1 });
```

Comment:
I don't see a reason why you need care about the previous status of void.

this function should just do an update.

if void? set it to true, otherwise unset should be sufficient?

Reply:
You can also update a revision status via the tag. I think your query is missing that part?

Reply:
![image](https://user-images.githubusercontent.com/11945337/133477004-e7f0f18c-6e03-4112-840a-45ed3c571f4c.png)


I lied about the v4 part... turns out only some v4 revision endpoints support tags and others don't.

going forward we want to support tags regardless.

Reply:
For the first one I thought that if the status provided is the same as the one in the db we shouldn't make the db call.

Reply:
sure, but there's not a lot of reason to do that. it's more costly to run 2 queries than 1